services/datcore-adapter/src/simcore_service_datcore_adapter/utils/client_base.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from dataclasses import dataclass
+from typing import Optional, Type
+
+import httpx
+from fastapi import FastAPI
+
+from .app_data import AppDataMixin
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class BaseServiceClientApi(AppDataMixin):
+    """
+    - wrapper around thin-client to simplify service's API calls
+    - sets endspoint upon construction
+    - MIME type: application/json
+    - processes responses, returning data or raising formatted HTTP exception
+    - helpers to create a unique client instance per application and service
+    """
+
+    client: httpx.AsyncClient
+    service_name: str
+    health_check_path: str = "/"
+
+    async def is_responsive(self) -> bool:
+        try:
+            resp = await self.client.get(self.health_check_path)
+            resp.raise_for_status()
+            return True
+        except (httpx.HTTPStatusError, httpx.RequestError) as err:
+            log.error("%s not responsive: %s", self.service_name, err)
+            return False
+
+
+# HELPERS -------------------------------------------------------------
+
+
+def setup_client_instance(
+    app: FastAPI,
+    api_cls: Type[BaseServiceClientApi],
+    api_baseurl,
+    service_name: str,
+    **extra_fields
+) -> None:
+    """Helper to add init/cleanup of ServiceClientApi instances in the app lifespam"""
+
+    assert issubclass(api_cls, BaseServiceClientApi)
+
+    def _create_instance() -> None:
+        api_cls.create_once(
+            app,
+            client=httpx.AsyncClient(base_url=api_baseurl),
```

Comment:
maybe add a default timeout here

Reply:
it is already set by default in httpx to 5 seconds. should I change it?

Reply:
Maybe make it explicit with an env var, so we can tweak on the go. Usually it comes in as handy.